### ISSUE TITLE
Allow retention of texture names and coordinates for non-Quake games

### ISSUE THREAD

**Issue by StomyPX** [ericwa/ericw-tools#430]
I've been using these tools for making my own game projects, but I'm limited to v2.0.0 alpha4 for now as I'm reliant on how `-notex` works before its changes introduced in alpha5.

What I specifically need is a way for `qbsp` to use loose `png` files in order to create correct texture coordinates and retain texture names in the final BSP but I don't need the actual mip textures themselves, just the names so my engine knows which images to load.

Having tried alpha8, `-notex` now just prunes the texnames completely while `-missing_textures_as_zero_size` instead borks the texture coordinates, making them useless. As both Trenchbroom and my engine only use the loose `png` files, I don't want to have to maintain `wad`s or `wal`s or any other extra files just for compilation.

I gather this isn't the intended use case for the tools, and for now I'm happy to stick to using alpha4 with `-notex`, but this setup has proven super convenient for me and I'd appreciate it if some switch were added to support this workflow in future releases.

**Comment by ericwa** [comment]
I think the specific thing that broke your workflow was we stopped reading png images in Q1 mode. This was the bug/discussion motivating that change: https://github.com/ericwa/ericw-tools/issues/404#issuecomment-1925939497

So if qbsp can’t locate the textures, it either writes 0x0 placeholders (if ‘-missing_textures_as_zero_size’ is in use) or omits the texture name entirely (the default). For a while we were writing 0x0 placeholders for missing textures by default without the opt-in flag, that turned out to be crashing Quake clients.

Would it be a feasible workaround to read the texture sizes from the png’s in your engine? That should work in 2.0.0-alpha8 if you use ‘-missing_textures_as_zero_size’. IIRC this is also how Q2 engines work since the texture sizes aren’t stored in the bsp in Q2?

**Comment by StomyPX** [comment]
> This was the bug/discussion motivating that change: https://github.com/ericwa/ericw-tools/issues/404#issuecomment-1925939497

Ah okay, well there's no point "fixing" this if it breaks something we actually want to work normally.

> Would it be a feasible workaround to read the texture sizes from the png’s in your engine? That should work in 2.0.0-alpha8 if you use ‘-missing_textures_as_zero_size’. IIRC this is also how Q2 engines work since the texture sizes aren’t stored in the bsp in Q2?

Yes actually! I have a separate util that repackages the final `.bsp` after running `vis` and `light`, I forgot that this is actually where the texcoords themselves are calculated. At the moment it uses the width/height supplied in the BspSurfaces packed into the `.bsp`, but there's nothing stopping me from just loading the pngs to check their width/height here manually. STB Image even has `stbi_info` which can get dimensions without actually decoding so there shouldn't even be much added overhead from disk I/O.

So in short, I'll close this because it's entirely a problem on my end that doesn't require the tools to change. Thank you!
